src/frontend/src/lib/api/icp-swap.api.ts

```diff
@@ -0,0 +1,161 @@
+import type { PoolData } from '$declarations/icp_swap_factory/icp_swap_factory.did';
+import { ICPSwapFactoryCanister } from '$lib/canisters/icp-swap-factory.canister';
+import { ICPSwapPoolCanister } from '$lib/canisters/icp-swap-pool.canister';
+import { ICP_SWAP_CANISTER_ID } from '$lib/constants/app.constants';
+import type {
+	ICPSwapAmountReply,
+	ICPSwapDepositWithdrawParams,
+	ICPSwapGetPoolParams,
+	ICPSwapGetUserUnusedBalanceParams,
+	ICPSwapQuoteParams,
+	ICPSwapQuoteSwapParams
+} from '$lib/types/api';
+import type { CanisterApiFunctionParams } from '$lib/types/canister';
+import { Principal } from '@dfinity/principal';
+import { assertNonNullish, isNullish } from '@dfinity/utils';
+
+export const getPool = async ({
+	identity,
+	token0,
+	token1,
+	fee,
+	canisterId = ICP_SWAP_CANISTER_ID
+}: CanisterApiFunctionParams<ICPSwapGetPoolParams>): Promise<PoolData> => {
+	assertNonNullish(canisterId, 'Missing pool canister ID');
+	assertNonNullish(identity, 'Identity is required');
+
+	const factory = await ICPSwapFactoryCanister.create({
+		identity,
+		canisterId: Principal.fromText(canisterId)
+	});
+	return factory.getPool({ token0, token1, fee });
+};
+
+const getPoolCanister = async ({
+	identity,
+	canisterId
+}: CanisterApiFunctionParams<{
+	canisterId: string;
+}>): Promise<ICPSwapPoolCanister> => {
```

Comment:
i think you don't need to pass here

```suggestion
}: CanisterApiFunctionParams): Promise<ICPSwapPoolCanister> => {
```

Reply:
As you explained offline, `canisterId` should be non-nullish:

`Omit< CanisterApiFunctionParams, 'canisterId'> & {canisterId: CanisterIdText;}`

Reply:
Resolved